server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -642,7 +643,7 @@ IndexMetadata buildAndValidateTemporaryIndexMetadata(
         tmpImdBuilder.setRoutingNumShards(routingNumShards);
         tmpImdBuilder.settings(aggregatedIndexSettings);
         tmpImdBuilder.system(isSystem);
-        addRemoteStoreCustomMetadata(tmpImdBuilder, true);
+        addRemoteStoreCustomMetadata(tmpImdBuilder, true, clusterState);
```

Comment:
We need to add this during non remote to remote store migration as well . Pls refer to `determineRemoteStoreCustomMetadataDuringMigration`  .

Reply:
We don't need to add any metadata to existing indexes during migration.